### ISSUE TITLE
DataCite workflow

### ISSUE THREAD

**Issue by cspayne** [uwlib-cams/uwlswd#45]

**Comment by cspayne** [comment]
@gerontakos @briesenberg07 

I currently have a draft rdf2datacite xsl script drawn up.
 I think it would be worth discussing the DataCite workflow and whether we want to investigate the DataCite API to reduce the manual labor in generating the DOI - I currently don't have the access required to investigate this myself.

**Comment by briesenberg07** [comment]
Copying [a comment](https://github.com/uwlib-cams/uwlswd/issues/25#issuecomment-1712362421) from @gerontakos which is on topic here:

> Creative solution needed for minting DOIs. At present, the DOI needs to be in the RDF/XML. However, at the time the DOI is created, neither the HTML/RDFa nor the DataCite XML metadata is created. As a result, we cannot complete the DOI registration: we don't have a URL (for the HTML) and we don't have metadata (for DataCite). So we need a solution for this. Currently we are doing it manually: create the draft DOI, insert in the RDF/XML, run the workflow scripts, the RDF/XML derivatives all contain the proper DOI, and the derivatives will be used to complete the DOI (manually), entering (in DataCite) the URL of the HTML and uploading the DataCite XML stored in uwlswd/DataCite. That's pretty clunky.

**Comment by gerontakos** [comment]
I can explore the DataCite API; however, if I have the luxury, I won't make it high priority, meaning I'll have an answer before September ends. Do I have that luxury? I can do it sooner if this is not an orthogonal piece of the workflow (I just wanted to use that word; I'm suggesting that if this piece of the workflow is a dependency for other pieces, I'll speed up my efforts in getting the API into the workflow, just let me know).

**Comment by cspayne** [comment]
> I can explore the DataCite API; however, if I have the luxury, I won't make it high priority, meaning I'll have an answer before September ends. Do I have that luxury? I can do it sooner if this is not an orthogonal piece of the workflow (I just wanted to use that word; I'm suggesting that if this piece of the workflow is a dependency for other pieces, I'll speed up my efforts in getting the API into the workflow, just let me know).

It's not a hold up for me! We did talk about testing the workflow on 94 vocabularies - which currently would need manually DOI generation. So that might get put on hold if you wanted to wait and see if we can get a better solution working.

**Comment by gerontakos** [comment]
Yes, here's an idea: after you're done with the "006" test, let me know and I'll run the workflow, using the Wiki instructions, on the NGL (I'm working on some edits today). Then I'll explore the API, communicate with you, then we'll start to think about the "007s" (i.e. the 94 vocabularies) for which David and I still have some work do, so they're not ready anyway.

**Comment by cspayne** [comment]
That sounds good! However, I don't currently have write access to the uwlswd_vocabs_marc_006_008 repository

**Comment by briesenberg07** [comment]
Just added you with write privileges @cspayne - let me know if you need more privileges!

**Comment by cspayne** [comment]
Thank you! I'm going to upload the "006" test now.

**Comment by cspayne** [comment]
I'm also going to update the wiki instructions to include running the rdf2datacite.xsl script

**Comment by gerontakos** [comment]
I have reviewed the "006" test. Here are several things I found, some of them are not errors. I didn't want to create an issue for everything. Some things may need to be broken out into separate issues.

006 form_of_material
**0. RDF/XML**
Faults in the RDF/XML
    a. incorrect dct:hasFormat values. Removed by TG (in RDF/XML only).
    b. Too bad order is not maintained. Old RDF/XML maintained the MARC standard's order, and that was helpful at times. I believe there's nothing we can do about that?
    c. Comments were removed. Could they be maintained? The order is important for comments; they precede a sequence to which they pertain. The way they're written, I don't think we can keep them and process the RDF/XML, given that order is not preserved. 

**_RDF/XML -->_**

**1. HTML/RDFa**
    _a. the html_
- Should not contain the triple dct:hasFormat=[itself]
- values in Predicate column are not dereferencing properly
    - dereference to url in prefix only.

_b. the RDFa_
- Superficially reviewed; looked ok.
    
_c. the schema.org_
- No language tag equivalents. Is that ok?
- Only one creator, a URI, is present; did we decide on literal values here?
- No alternateName present (i.e. alternative title)
- Publisher: also only an IRI. I believe decision is pending on this?
- "@type" : "Dataset" --> I believe decision is pending on this?
- contributor property spelled contributer.
- Only contributors with IRIs are present. Incorrect?
- No isBasedOn (i.e. dct:source) present.
- In 006/form_of_material.html, "description" : "VALUE MISSING" seems incorrect

**2. JSON-LD**
- When RDF is indented, new lines show up in JSON-LD (i.e. \n)
    - solution: RDF/XML should not be indented.

**3. N-Triples (same indentation problem; otherwise OK)**

**4. Turtle (same indentation problem)**

**5. Datacite XML**
- Value of resourceType is not correct (should be some phrase; I don't think we've resolved how to do this.)
- //title/@titleType for alt title should be AlternativeTitle not Alternative Title.
- //relatedIdentifiers/relatedIndentifier should be //relatedIdentifiers/relatedIdentifiers	
- //relatedIdentifier/@relationType="isDerivedFrom" should be //relatedIdentifier/@relationType="IsDerivedFrom"

**Comment by cspayne** [comment]
@gerontakos @briesenberg07 

Thank you for this feedback Theo! 
I went ahead and started a [discussion](https://github.com/uwlib-cams/uwlswd/discussions/52) and added some comments and questions.
